src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1262,6 +1274,77 @@ .maxstack  1
     IL_0000:  ldarg.0
     IL_0001:  call       ""object..ctor()""
     IL_0006:  ret
+}");
+        }
+
+        [Fact]
+        public void TestNullCheckedIterator()
+        {
+            var source = @"
+using System.Collections.Generic;
+class Iterators
+{
+    IEnumerable<char> GetChars(string s!)
+    {
+        foreach (var c in s)
+        {
+            yield return c;
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

Please test with an empty iterator:
```C#
static IEnumerable<char> GetChars(string s!)
{
    yield break;
}
```
Are we throwing the exception on the call to get the `IEnumerable<char>`, or a subsequent call to `GetEnumerator()`, or later?